tsconfig.json

```diff
@@ -0,0 +1,60 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    // "lib": [],                             /* Specify library files to be included in the compilation. */
+    "allowJs": true,                         /* Allow javascript files to be compiled. */
```

Comment:
Are we compiling JS via `tsc`? Does this option need to be enabled?

Reply:
For the objective of this PR it isn't needed, so turning it off.

Strictly speaking, I don't think its using `tsc` (still very new to this ecosystem). `rollup-plugin-typescript` grabs and uses the API from `typescript/lib/typescript.js`. And looks like `tsc` is purely the CLI tool to use the same APIs.

Reply:
Ah, I didn't mean `tsc` directly. As of a couple of weeks ago, there are two options for transforming TS code to JS code: the typescript compiler (which actually compiles, and emits JS), and babel 7 (which just knows how to eat the non-javascript parts, leaving only JS, but no type-checking whatsoever).

In this rollup plugin the contents of `tsconfig.json` are read

https://github.com/rollup/rollup-plugin-typescript/blob/7f553800b90fc9abdcd13636b17b4824459d3960/src/options.js#L40-L52

which is then passed to the typescript compiler

https://github.com/rollup/rollup-plugin-typescript/blob/7f553800b90fc9abdcd13636b17b4824459d3960/src/index.js#L92-L96

so whatever we do here is _as if_ we were just running `tsc` on the command line.